src/machine-c-api.cpp

```diff
@@ -352,6 +364,19 @@ cm_error cm_log_step_uarch(cm_machine *m, int32_t log_type, const char **log) tr
     return cm_result_failure();
 }
 
+cm_error cm_verify_step(const cm_hash *root_hash_before, const char *log_filename, uint64_t mcycle_count,
+    const cm_hash *root_hash_after) try {
+    if (log_filename == nullptr) {
+        throw std::invalid_argument("invalid log_filename");
+    }
+    const cartesi::machine::hash_type cpp_root_hash_before = convert_from_c(root_hash_before);
+    const cartesi::machine::hash_type cpp_root_hash_after = convert_from_c(root_hash_after);
+    cartesi::machine::verify_step(cpp_root_hash_before, log_filename, mcycle_count, cpp_root_hash_after);
```

Comment:
This has to receive a `const cm_machine *m`. If it is nullptr, then use the static method directly. If it is not null, convert to `cpp_m` and call `cpp_m->verify_step()`